core/prisma/schema.prisma

```diff
@@ -295,19 +306,98 @@ enum MemberRole {
 }
 
 model MemberGroup {
-  id          String     @id @default(dbgenerated("gen_random_uuid()"))
-  role        MemberRole @default(editor)
-  community   Community  @relation(fields: [communityId], references: [id], onDelete: Cascade)
+  id          String    @id @default(dbgenerated("gen_random_uuid()"))
+  community   Community @relation(fields: [communityId], references: [id], onDelete: Cascade)
   communityId String
-  createdAt   DateTime   @default(now())
-  updatedAt   DateTime   @default(now()) @updatedAt
+  createdAt   DateTime  @default(now())
+  updatedAt   DateTime  @default(now()) @updatedAt
 
-  permissions Permission[]
-  users       User[]
+  permissions          Permission[]
+  users                User[]
+  formMemberships      FormMembership[]
+  communityMemberships CommunityMembership[]
+  pubMemberships       PubMembership[]
+  stageMemberships     StageMembership[]
 
   @@map(name: "member_groups")
 }
 
+model CommunityMembership {
+  id            String       @id @default(dbgenerated("gen_random_uuid()"))
+  role          MemberRole
+  community     Community    @relation(fields: [communityId], references: [id], onDelete: Cascade)
+  communityId   String
+  user          User?        @relation(fields: [userId], references: [id], onDelete: Cascade)
+  userId        String?
+  memberGroup   MemberGroup? @relation(fields: [memberGroupId], references: [id], onDelete: Cascade)
+  memberGroupId String?
```

Comment:
ideally the type generated by `kanel` would be a discriminated union having either the `userId` or the `memberGroupId`. I'm not sure that's possible without a custom prerender hook.
 Working with these mysterious nullable columns is always kind of annoying to me, so i'm very glad you added the `check` constraint!

Reply:
yeah it would be great if the generated type had a discriminated union. at the very least, i think we should be able to manually construct a type from the generated one that enforces our constraint. for now, i've just been omitting `memberGroupId` from the generated type since we aren't using it anywhere yet